statistics/handle/lockstats/lock_stats.go

```diff
@@ -82,26 +82,88 @@ func AddLockedTables(exec sqlexec.RestrictedSQLExecutor, tids []int64, pids []in
 	}
 
 	// Insert related partitions while don't warning duplicate partitions.
-	checkedTables = GetLockedTables(lockedTables, pids...)
+	lockedPartitions := GetLockedTables(lockedTables, pids...)
 	for _, pid := range pids {
-		if _, ok := checkedTables[pid]; !ok {
+		if _, ok := lockedPartitions[pid]; !ok {
 			if err := insertIntoStatsTableLocked(ctx, exec, pid); err != nil {
 				return "", err
 			}
 		}
 	}
 
-	msg := generateSkippedTablesMessage(tids, skippedTables, lockAction, lockedStatus)
+	msg := generateSkippedMessage(tids, skippedTables, lockAction, lockedStatus)
 	// Note: defer commit transaction, so we can't use `return nil` here.
 	return msg, err
 }
 
-func generateSkippedTablesMessage(tids []int64, dupTables []string, action, status string) string {
-	if len(dupTables) > 0 {
-		tables := strings.Join(dupTables, ", ")
+// AddLockedPartitions add locked partitions id to store.
+// If the whole table is locked, then skip all partitions of the table.
+// - exec: sql executor.
+// - tid: table id of which will be locked.
+// - tableName: table name of which will be locked.
+// - pidNames: partition ids of which will be locked.
+// Return the message of skipped tables and error.
+func AddLockedPartitions(
```

Comment:
Is it possible to merge this function with `AddLockedTables`?

Reply:
It would make the AddLockedTables more complicated and unreadable.